### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by MitraDarja**
First fixes like typos, layout and death links.

### CODE REVIEW

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp [hunk 1]
```diff
@@ -386,7 +386,7 @@ typename range_type::iterator insert_gap(range_type & rng,
 
 /*!\brief An implementation of seqan3::AlignedSequence::erase_gap for ranges with the corresponding
  *        member function erase_gap(it).
- * \ingroup seqan3::AlignedSequence
+ * \ingroup aligned_sequence
```
Comment (smehringer) [inline include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp]: why this change? I am pretty sure this is the correct submodule to group it to.

File: include/seqan3/argument_parser/detail/format_man.hpp [hunk 2]
```diff
@@ -30,7 +30,7 @@ namespace seqan3::detail
  * provide meta information, positional options, etc. in the correct order.
  * In addition the needed order would be different from the parse format.
  * Thus the calls are stored (parser_set_up_calls and positional_option_calls)
- * and only evaluated when calling seqan3::format_help::parse.
+ * and only evaluated when calling  seqan3::detail::format_help_base::parse.
```
Comment (smehringer) [inline include/seqan3/argument_parser/detail/format_man.hpp]:
```suggestion
 * and only evaluated when calling seqan3::detail::format_help_base::parse.
```
:nail_care:

File: include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp (hunk 1 above)
Comment (MitraDarja) [reply include/seqan3/alignment/aligned_sequence/aligned_sequence_concept.hpp]: http://docs.seqan.de/seqan/3-master-user/group__aligned__sequence.html#ga0c6fc7bb7f0554a0085e521b214c812c
Look at erase_gap 3/4, there are :: in the end of the description, this change gets rid of them.
